modules/terraform/output.go

```diff
@@ -49,6 +50,131 @@ func OutputRequiredE(t testing.TestingT, options *Options, key string) (string,
 	return out, nil
 }
 
+func parseListOfMaps(l []interface{}) ([]map[string]interface{}, error) {
+	var result []map[string]interface{}
+
+	for _, v := range l {
+		m, err := parseMap(v.(map[string]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, m)
+	}
+
+	return result, nil
+
+}
+
+// parseMap takes a map of interfaces and parses the types. Is also recursive for nested objects.
+func parseMap(m map[string]interface{}) (map[string]interface{}, error) {
+
+	var result map[string]interface{}
+	result = make(map[string]interface{})
+
+	for k, v := range m {
+		switch v.(type) {
+		case map[string]interface{}:
+			nestedMap, err := parseMap(v.(map[string]interface{}))
+			if err != nil {
+				return nil, err
+			}
+			result[k] = nestedMap
+		case []interface{}:
+			nestedList, err := parseListOfMaps(v.([]interface{}))
+			if err != nil {
+				return nil, err
+			}
+			result[k] = nestedList
+		default:
+			// Test for int conversion
+			testInt, err := strconv.ParseInt((fmt.Sprintf("%v", v)), 10, 0)
```

Comment:
What about float? Bool?

Reply:
I thought about adding support for the other data types as well, but the `OutputMap` and `OutputList` functions also only return string versions of their values. I will work on a new commit to support this right away.